temporalnexus/token.go

```diff
@@ -0,0 +1,79 @@
+// The MIT License
+//
+// Copyright (c) 2025 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package temporalnexus
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+type operationTokenType int
+
+const (
+	operationTokenTypeWorkflowRun = operationTokenType(1)
+)
+
+// workflowRunOperationToken is the decoded form of the workflow run operation token.
+type workflowRunOperationToken struct {
+	// Type of the operation. Must be operationTypeWorkflowRun.
+	Type          operationTokenType `json:"t"`
+	NamespaceName string             `json:"ns"`
+	WorkflowID    string             `json:"wid"`
+}
+
+func generateWorkflowRunOperationToken(namespace, workflowID string) (string, error) {
+	token := workflowRunOperationToken{
+		Type:          operationTokenTypeWorkflowRun,
+		NamespaceName: namespace,
+		WorkflowID:    workflowID,
+	}
+	data, err := json.Marshal(token)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal workflow run operation token: %w", err)
+	}
+	return base64.URLEncoding.WithPadding(base64.NoPadding).EncodeToString(data), nil
+}
+
+func loadWorkflowRunOperationToken(data string) (workflowRunOperationToken, error) {
```

Comment:
During the meeting yesterday @cretz suggested we reserve `v` for a versioning field and fail if it is present. Did we decided not to do that?

Reply:
Coming in the next commit.